Islandora/islandora_solr_search
# Islandora 1889 Escapeing search terms, preventing error searching solr index.

**JIRA Ticket**: (https://jira.duraspace.org/browse/ISLANDORA-1889)

# What does this Pull Request do?
When preforming an advanced search on the islandora advanced search block, special characters (such as '[' or ']') will throw an 'Error searching Solr index "400" Status: Bad Request'.

# What's new?
Implements the [islandora_solr_facet_escape()|https://github.com/Islandora/islandora_solr_search/blob/7.x/includes/utilities.inc#L165] utility function on the search term prior to query execution.

# How should this be tested?
- Navigate to the advanced search block.
- Preform a search query with an invalid character (such as 'Lobster[' or 'lobster{{}'). Results pages should throw a 400 error.

# Steps to reproduce
- Navigate to the advanced search block.
- Preform a search query with an invalid character (such as 'Lobster[' or 'lobster{{}'). Results pages should not throw a 400 error.

# Interested parties
@Islandora/7-x-1-x-committers
